lib/cstruct.ml

```diff
@@ -296,13 +296,23 @@ end
 let len t =
   t.len
 
-let lenv = function
-  | []  -> 0
-  | [t] -> len t
-  | ts  -> List.fold_left (fun a b -> len b + a) 0 ts
+(** [sum_lengths ~caller acc l] is [acc] plus the sum of the lengths
+    of the elements of [l].  Raises [Invalid_argument caller] if 
+    arithmetic overflows. *)
+let rec sum_lengths_aux ~caller acc = function
+  | [] -> acc
+  | h :: [] -> len h + acc
```

Comment:
It can overflow, but the result will be negative if it does, and subsequently caught by the caller.  But this is probably too subtle and we should just check directly here.